sdcm/mgmt.py

```diff
@@ -4,10 +4,292 @@
+            progress = self.progress
+        return self.status in list_status
+
+    def wait_for_status(self, list_status, check_task_progress=True):
+        text = "Waiting until task: {} reaches status of: {}".format(self.id, list_status)
+        is_status_reached = False
+        is_status_reached = wait.wait_for(func=self.is_status_in_list, step=60,
+                                          text=text, list_status=list_status, check_task_progress=check_task_progress, timeout=3600)
+        return is_status_reached
+
+    def wait_for_status_done(self):
+        """
+        Wait and report that task is eventually reaching a 'final' status. meaning one of: done/error/stopped
+        :return:
+        """
+        logger.debug("Waiting for task: {} to be done..".format(self.id))
+        cur_status = self.status
+        if cur_status == TaskStatus.DONE:
+            return True
+        if cur_status in [TaskStatus.ERROR, TaskStatus.STOPPED]:
+            return False
+
+        if cur_status == TaskStatus.NEW:
+            logger.debug("Waiting for task: {} to start..".format(self.id))
+            list_status = [TaskStatus.RUNNING, TaskStatus.ERROR, TaskStatus.DONE]
+            res = self.wait_for_status(list_status=list_status)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status".format(self.id))
+            cur_status = self.status
+            if cur_status == TaskStatus.ERROR:
+                return False
+
+        if cur_status == TaskStatus.RUNNING:
+            logger.debug("Waiting for task: {} to finish running..".format(self.id))
+            list_status = [TaskStatus.DONE, TaskStatus.ERROR]
+            res = self.wait_for_status(list_status=list_status)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status {}".format(self.id, list_status))
+            cur_status = self.status
+            if cur_status == TaskStatus.ERROR:
+                return False
+        final_state = self.status
+        logger.debug("Task: {} final state is: {}".format(self.id, str(final_state)))
+        return self.status == TaskStatus.DONE
+
+class ManagerCluster(ScyllaLogicObj):
+
+    def __init__(self, mgr_tool, cluster_id):
+        if not mgr_tool:
+            raise ScyllaManagerError("Cannot create a Manager Cluster where no 'manager' host parameter is given")
+        ScyllaLogicObj.__init__(self, id=cluster_id)
+        self.mgr_tool = mgr_tool
+
+    def create_repair_task(self):
+        cmd = "repair -c {}".format(self.id)
+        res = self.mgr_tool.run_sctool_cmd(cmd=cmd)
+        if not res:
+            raise ScyllaManagerError("Unknown failure for sctool {} command".format(cmd))
+
+        if "no matching units found" in res.stderr:
+            raise ScyllaManagerError("Manager cannot run repair where no keyspace exists.")
+
+        if 'repair' not in res.stdout:
```

Comment:
Is it the prefix of the task ID or something?
Can you please add comment above that describe/example the output we expect?

Reply:
added comment: 
# expected result output is to have a format of: "repair/2a4125d6-5d5a-45b9-9d8d-dec038b3732d"

Reply:
thx